etl-pipeline/managers/s3.py

```diff
@@ -28,39 +27,9 @@ def __init__(self):
             endpoint_url=os.environ.get('S3_ENDPOINT_URL', None)
         )
 
-    def store_pdf_manifest(self, record: Record, bucket_name, flags=FileFlags(reader=True), path: str=None):
+    def store_pdf_manifest(self, record: Record, bucket_name, flags=FileFlags(reader=True), path: str=None):  
         
-        if record.source == 'doab':
-            link_manager = DOABLinkManager(record)
-
-            link_manager.parse_links()
-
-            print(f'Length of manifests: {len(link_manager.manifests)}')
-            
-            for manifest in link_manager.manifests:
-                manifest_path, manifest_json = manifest
-                self.create_manifest_in_s3(manifest_path=manifest_path, manifest_json=manifest_json, bucket=bucket_name)
-
-                print(f'manifest_path={manifest_path}')
-                record.has_part.insert(0, str(Part(
-                    index=1,
-                    url=manifest_path,
-                    source=record.source,
-                    file_type='application/webpub+json',
-                    flags=str(flags)
-                )))
-
-            for epub_link in link_manager.epub_links:
-                epub_path, epub_uri = epub_link
-                record.has_part.append(str(Part(
-                    index=1,
-                    url=epub_path,
-                    source=Source.LOC.value,
-                    file_type='application/epub+zip',
-                    flags=str(FileFlags(download=True)),
-                    source_url=epub_uri
-                )))
-        else:
+        if record.source != Source.DOAB.value:
```

Comment:
I think we should steer away from having source-specific logic in any of our managers and the record pipeline.

Reply:
Ok that's a good conditional to make sure the manifest isn't being created again. That was my original intent for adding the `if record.source != Source.DOAB.value` conditional.